lib/src/types.ts

```diff
@@ -128,22 +145,26 @@ export interface ObjectPropertiesType {
 export interface ObjectType {
   kind: TypeKind.OBJECT;
   properties: Array<ObjectPropertiesType>;
+  schemaProps?: SchemaProp[];
 }
 
 export interface ArrayType {
   kind: TypeKind.ARRAY;
   elementType: Type;
+  schemaProps?: SchemaProp[];
 }
 
 export interface UnionType {
   kind: TypeKind.UNION;
   types: Type[];
   discriminator?: string;
+  schemaProps?: SchemaProp[];
 }
 
 export interface IntersectionType {
   kind: TypeKind.INTERSECTION;
   types: Type[];
+  schemaProps?: SchemaProp[];
```

Comment:
I don't think `UnionType` and `IntersectionType` have any need for `schemaProps` - doesn't look like OpenAPI supports any props for `allOf` or `oneOf`.

Reply:
For `allOf` or `oneOf`, we can see that it inherits from `SchemaObjectBase`, so it includes the following attributes:
- `title`
- `example`
- `deprecated` (only for OAS3)

For `UnionType`, we will use it for `String` enum for example.

I added a config table for all combinations in `lib/src/parsers/schemaprop-parser.ts`
Based on the list, there is new unit tests in `lib/src/generators/openapiX/__spec-examples__/contract-with-schemaprops.ts`
Then, I added also the OAS2 implemntation for SchemaProperties.
Let me know.

Reply:
👍 thanks @Thodin3